functions/utils/getAsset.ts

```diff
@@ -0,0 +1,25 @@
+import { AssetDocument } from '../../src/graphql/data/__generated__/types-and-hooks'
+import { getApolloClient } from './getApolloClient'
+
+export default async function getAsset(collectionAddress: string, tokenId: string, url: string) {
+  const client = getApolloClient()
+  const { data } = await client.query({
+    query: AssetDocument,
+    variables: {
+      address: collectionAddress,
+      filter: {
+        tokenIds: [tokenId],
+      },
+    },
+  })
+  const asset = data?.nftAssets?.edges[0]?.node
+  if (!asset) {
+    return undefined
+  }
+  const formattedAsset = {
+    title: asset.name ? asset.name : asset.collection?.name + ' #' + asset.tokenId,
```

Comment:
you should check for `asset.collection?.name` before using it, otherwise you'll get a leading space if it's not there.